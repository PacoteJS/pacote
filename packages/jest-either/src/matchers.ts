export interface AsymmetricMatcher {
  asymmetricMatch(other: any): boolean
}

export function isAsymmetricMatcher(
  matcher: any
): matcher is AsymmetricMatcher {
  return typeof matcher.asymmetricMatch === 'function'
}
