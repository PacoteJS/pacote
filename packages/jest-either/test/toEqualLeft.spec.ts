import { left, right } from 'fp-ts/lib/Either'
import matchers from '../src'

expect.extend(matchers)

describe('.toEqualLeft()', () => {
  test('passes when object is a left', () => {
    expect(left({ a: { b: 'left' } })).toEqualLeft({ a: { b: 'left' } })
  })

  test('fails when object is not equal', () => {
    expect(() =>
      expect(left({ a: { b: 'left' } })).toEqualLeft('wrong')
    ).toThrowErrorMatchingSnapshot()
  })

  test('fails when object is a right', () => {
    expect(() =>
      expect(right('right')).toEqualLeft('right')
    ).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toEqualLeft()', () => {
  test('passes when object is a right', () => {
    expect(right('right')).not.toEqualLeft('left')
  })

  test('passes when object is not equal', () => {
    expect(left({ a: { b: 'left' } })).not.toEqualLeft('wrong')
  })

  test('fails when object is a left', () => {
    expect(() =>
      expect(left('left')).not.toEqualLeft('left')
    ).toThrowErrorMatchingSnapshot()
  })
})
