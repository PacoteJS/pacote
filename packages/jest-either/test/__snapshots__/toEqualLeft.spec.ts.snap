// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toEqualLeft() fails when object is a left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toEqualLeft([22m[32mexpectedLeft[39m[2m)[22m

Expected Either not to equal left:
  [32m\\"left\\"[39m

Received left:
  [31m\\"left\\"[39m"
`;

exports[`.toEqualLeft() fails when object is a right 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toEqualLeft([22m[32mexpectedLeft[39m[2m)[22m

Expected Either to equal left:
  [32m\\"right\\"[39m

Received right."
`;

exports[`.toEqualLeft() fails when object is not equal 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toEqualLeft([22m[32mexpectedLeft[39m[2m)[22m

Expected Either to equal left:
  [32m\\"wrong\\"[39m

Received left:
  [31m{\\"a\\": {\\"b\\": \\"left\\"}}[39m"
`;
